### ISSUE TITLE
hyperparmeter search

### ISSUE THREAD

**Issue by KimDaeUng** [boostcampaitech2/model-optimization-level3-nlp-14#1]
## `tune.py` 수정
- 이미지 입력 사이즈 설정 부분 수정 : data_config 에서의 이미지 사이즈와 model_conifg 에서의 이미지 사이즈 통일되도록 변경
- Early Stopping 기능 추가(src/utils/torch_utils ) : trainer.py 에서 patience 와 delta 값 수정 가능
- hyperparameter 탐색 옵션에 optimizer, learning rate 추가
- 샘플링 기능 추가 : data_config["SAMPLING_RATIO"] 로 전체 데이터셋 대비 몇 %의 데이터를 샘플링할 지 설정

## 탐색 방법
- 절차 : 일부 데이터셋 만으로 초반 학습 결과를 보고 떡잎이 될 후보 모델 선정(`exp/latest` 에 `best.ts` 란 이름으로 저장됩니다)하고 이걸 `train.py` 로 전체 데이터로 다시 학습하여 제출합니다.
탐색 시간을 줄이기 위해 아래 조건은 통일
- train set의 40%만 샘플링
- epoch 50
- 각자 설정할 부분
  - learning rate
  - optimizer
  - 각 모듈(m1-m7)의 최소, 최대 반복 횟수 범위 조절
  - 모듈을 일부 삭제 하거나 추가
  - 기타 코드에서 제공된 기본값의 범위 조절
- 실험 결과는 wandb에 공유 부탁드립니다(중복 케이스 무시)

## TODO
- @Amber-Chaeeunk : cross entropy loss weight 추가
- @hrxorxm : train 코드에 wandb 추가
